modernizer-annotation-processor/src/main/java/org/gaul/modernizer_annotation_processor/ModernizerAnnotationProcessor.java

```diff
@@ -0,0 +1,331 @@
+
+    @Override
+    public final SourceVersion getSupportedSourceVersion() {
+        return SourceVersion.latest();
+    }
+
+    private class AnnotatedElements {
+        private List<String> annotatedClasses;
+        private List<String> annotatedMethods;
+
+        AnnotatedElements() {
+            annotatedClasses = new ArrayList<String>();
+            annotatedMethods = new ArrayList<String>();
+        }
+
+        public List<String> getAnnotatedClasses() {
+            return annotatedClasses;
+        }
+
+        public void setAnnotatedClasses(List<String> annotatedClasses) {
+            this.annotatedClasses = annotatedClasses;
+        }
+
+        public List<String> getAnnotatedMethods() {
+            return annotatedMethods;
+        }
+
+        public void setAnnotatedMethods(List<String> annotatedMethods) {
+            this.annotatedMethods = annotatedMethods;
+        }
+    }
+
+    @Override
+    public final boolean process(
+        Set<? extends TypeElement> annotations,
+        RoundEnvironment roundEnv
+    ) {
+        for (TypeElement annotation : annotations) {
+            AnnotatedElements annotatedElements =
+                getAnnotatedElements(roundEnv, annotation);
+            if (!(annotatedElements.getAnnotatedClasses().isEmpty() &&
+                annotatedElements.getAnnotatedMethods().isEmpty())) {
+                File outputDir = getOutputDirectory();
+                outputDir.mkdirs();
+                makeFile(new File(outputDir,
+                    ModernizerAnnotationUtils.IGNORE_CLASSES_FILE_NAME),
+                    annotatedElements.getAnnotatedClasses());
+                makeFile(new File(outputDir,
+                    ModernizerAnnotationUtils.IGNORE_METHODS_FILE_NAME),
+                    annotatedElements.getAnnotatedMethods());
+            }
+        }
+        return true;
+    }
+
+    private AnnotatedElements getAnnotatedElements(
+        RoundEnvironment roundEnv,
+        TypeElement annotation
+    ) {
+        AnnotatedElements elements = new AnnotatedElements();
+        List<String> annotatedClasses = new ArrayList<String>();
+        List<String> annotatedMethods = new ArrayList<String>();
+        for (Element element : roundEnv.getElementsAnnotatedWith(annotation)) {
+            List<String> warnings = Arrays.asList(
+                    element.getAnnotation(SuppressWarnings.class).value());
+            if (warnings.contains("modernizer")) {
+                if (element.getKind().isClass()) {
+                    annotatedClasses.add(
+                        getClassHeaderRegex(element));
+                } else if (element.getKind().equals(ElementKind.METHOD) ||
+                    element.getKind().equals(ElementKind.CONSTRUCTOR)) {
+                    annotatedMethods.add(getMethodRepresentation(element));
+                }
+            }
+        }
+        elements.setAnnotatedClasses(annotatedClasses);
```

Comment:
Can you have the constructor take both of these and remove the setter? Then we could just do `return new AnnotatedElements(...)` here